### ISSUE TITLE
grails is using wrong tenant id when a where clause is used in a query

### ISSUE THREAD

**Issue by jvesouza** [grails/grails-core#10825]
### Steps to Reproduce

1. Launch the sample application.
2. Access the following url: (http://localhost:8080/plan).

### Expected Behaviour

The JSON returned by the application should have the same value in the property 'find' and in the property 'find with selection'.

### Actual Behaviour

The JSON returned has different values in the two properties.

### Environment Information

- **Operating System**: linux
- **Grails Version:** 3.1.1
- **JDK Version:** openjdk version "1.8.0_144"
- **Container Version (If Applicable):** N/A
### Example Application
- (https://github.com/jvesouza/demomt)

Basically the application implements a TenantResolver that always returns a fixed tenant id and a service that executes the following  queries in a table.
1. Plan p1 = Plan.findAll()[0]
2. Plan p2 = Plan.findAll({ id > 0 })[0]

After execute the two queries, the service returns a JSON containning the value of the attribute description present in p1 and p2.

It seems that when findAll with id > 0 is executed, the default connection is used instead of the connection pointed by the current tenant id.
